src/DeploymentTemplate.ts

```diff
@@ -323,6 +334,70 @@ export class DeploymentTemplate {
         return functionCounts;
     }
 
+    /**
+     * Gets info about schema usage, useful for telemetry
+     */
+    public getResourceUsage(): Histogram {
+        const resourceCounts = new Histogram();
+
+        // Collect all resources used
+        const resources: Json.ArrayValue | null = this._topLevelValue ? Json.asArrayValue(this._topLevelValue.getPropertyValue(templateKeys.resources)) : null;
+        if (resources) {
+            traverseResources(resources, undefined);
+        }
+
+        return resourceCounts;
+
+        function traverseResources(resourcesObject: Json.ArrayValue, parentKey: string | undefined): void {
+            for (let resource of resourcesObject.elements) {
+                const resourceObject = Json.asObjectValue(resource);
+                if (resourceObject) {
+                    const resourceType = Json.asStringValue(resourceObject.getPropertyValue(templateKeys.resourceType));
+                    const apiVersion = Json.asStringValue(resourceObject.getPropertyValue(templateKeys.resourceApiVersion));
+                    if (resourceType && apiVersion) {
```

Comment:
Can we also get data when apiVersion is blank?  I.e. when someone is using an apiProfile?

Reply:
Sure.